File: AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/applovin/DemoMenuItem.java
package com.applovin.enterprise.apps.demoapp.ads.applovin;

import android.content.Intent;

public class DemoMenuItem
{
    private final String title;
    private final Intent intent;

    public DemoMenuItem(final String title, final Intent intent)
    {
        this.title = title;
        this.intent = intent;
    }

    public String getTitle()
    {
        return title;
    }

    public Intent getIntent()
    {
        return intent;
    }
}

File: AppLovin MAX Demo App - Kotlin/app/src/main/java/com/applovin/enterprise/apps/demoapp/data/main/DemoMenuItem.kt
package com.applovin.enterprise.apps.demoapp.data.main

import android.content.Intent

/**
 * A [ListItem] representing an ad type on the main screen.
 * <p>
 * Created by Harry Arakkal on 09/17/2019.
 */
data class DemoMenuItem(val title: String, val intent: Intent?, val runnable:Runnable?, override val type: Int = ListItem.AD_ITEM) : ListItem
{
    constructor(title: String, intent: Intent) : this(title, intent, null)
    constructor(title: String, runnable:Runnable) : this(title, null, runnable)
}
